### DISCUSSION TITLE
terser-webpack-plugin throw error

### DISCUSSION THREAD

**Discussion by yangwangliang** [webpack/webpack#15937] [General]
# Bug report

**What is the current behavior?**
 local project running status is good，when i build project terser throw error Unexpected token: punc ({)

**If the current behavior is a bug, please provide the steps to reproduce.**
  this is log
    0 info it worked if it ends with ok
    1 verbose cli [ '/usr/local/bin/node', '/usr/local/bin/npm', 'run', 'build' ]
    2 info using npm@6.14.7
    3 info using node@v14.7.0
    4 verbose run-script [ 'prebuild', 'build', 'postbuild' ]
    5 info lifecycle Erp@1.0.0~prebuild: Erp@1.0.0
    6 info lifecycle Erp@1.0.0~build: Erp@1.0.0
    7 verbose lifecycle Erp@1.0.0~build: unsafe-perm in lifecycle true
    8 verbose lifecycle Erp@1.0.0~build: PATH: /usr/local/lib/node_modules/npm/node_modules/npm-lifecycle/node-gyp-bin:/Users/yangwangliang/workdata/release/Public/Erp/node_modules/.bin:/usr/local/opt/php@8.0/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/usr/local/opt/php@8.0/bin
    9 verbose lifecycle Erp@1.0.0~build: CWD: /Users/yangwangliang/workdata/release/Public/Erp
    10 silly lifecycle Erp@1.0.0~build: Args: [ '-c', 'webpack --progress --config build/webpack.pro.js' ]
    11 silly lifecycle Erp@1.0.0~build: Returned: code: 1  signal: null
    12 info lifecycle Erp@1.0.0~build: Failed to exec build script
    13 verbose stack Error: Erp@1.0.0 build: `webpack --progress --config build/webpack.pro.js`
    13 verbose stack Exit status 1
    13 verbose stack     at EventEmitter.<anonymous> (/usr/local/lib/node_modules/npm/node_modules/npm-lifecycle/index.js:332:16)
    13 verbose stack     at EventEmitter.emit (events.js:314:20)
    13 verbose stack     at ChildProcess.<anonymous> (/usr/local/lib/node_modules/npm/node_modules/npm-lifecycle/lib/spawn.js:55:14)
    13 verbose stack     at ChildProcess.emit (events.js:314:20)
    13 verbose stack     at maybeClose (internal/child_process.js:1051:16)
    13 verbose stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:287:5)
    14 verbose pkgid Erp@1.0.0
    15 verbose cwd /Users/yangwangliang/workdata/release/Public/Erp
    16 verbose Darwin 21.2.0
    17 verbose argv "/usr/local/bin/node" "/usr/local/bin/npm" "run" "build"
    18 verbose node v14.7.0
    19 verbose npm  v6.14.7
    20 error code ELIFECYCLE
    21 error errno 1
    22 error Erp@1.0.0 build: `webpack --progress --config build/webpack.pro.js`
    22 error Exit status 1
    23 error Failed at the Erp@1.0.0 build script.
    23 error This is probably not a problem with npm. There is likely additional logging output above.
    24 verbose exit [ 1, true ]

**What is the expected behavior?**

**Other relevant information:**
webpack version:  5.51.1
Node.js version: 14.7.0
Operating System: win7 or macos 12.4
Additional tools:

**Comment by vankop** [comment]
please create reproducible repo
_______

Probably bug in `terser` itself ( or it doesn't support some syntax ), we need code snippet to check this
